src/usdb_syncer/resource_dl.py

```diff
@@ -119,14 +152,34 @@ def download_video(
         path_stem: the target on the file system *without* an extension
 
     Returns:
-        the extension of the successfully downloaded file or None
+        DownloadResult with the extension of the downloaded file if successful
     """
     ydl_opts = _ytdl_options(
         options.ytdl_format(), browser, path_stem, options.rate_limit
     )
-    if filename := _download_resource(ydl_opts, resource, logger):
-        return os.path.splitext(filename)[1][1:]
-    return None
+    return _download(resource, ydl_opts, logger)
+
+
+def _download(resource: str, options: YtdlOptions, logger: Log) -> ResourceDLResult:
```

Comment:
It's great that now a type-safe result object is returned!
However, I don't understand the necessity of creating an exception in `_download_resource` that is then immediately replaced by an enum variant in `_download`. Why not directly return the `ResourceDLError` (or `ResourceDLResult`)?
Also, logging these exception is not helpful if they don't have a custom `__str__` method or similar.

Reply:
Thanks for the detailed review and your comments, really appreciated! I think I've considered them all and changed the code accordingly.

Reply:
Looks great! The exception classes can be removed now, right?

Reply:
Yes, I forgot to remove them. Would have thought that pylint/mypy would have pointed out unused classes... They're removed now.

Reply:
I think it only works for private names as public ones could be part of a public API.